### ISSUE TITLE
How to build TerraMind encdec model pretrained

### ISSUE THREAD

**Issue by fgiauna** [IBM/terramind#46]
Hello,
In terratorch with Prithvi it was possible to build PrithviMAE instead of PrithviViT by passing `"backbone_encoder_only": False` together with `"backbone_pretrained": True` to model_args. I'm trying to do something similar with TerraMind (the generic class from terramind.py, not TerraMindViT) as for my use case I will need both the encoder and decoder (the TerraMind one and not an additional decoder like UNet). I'm just trying to fine-tune the model on a downstream task, for instance burn-scars.

I tried to use `terramind_v1_base_encdec` but it requires encoder and decoder embeddings dict. I saw that TerraMindGeneration builds on TerraMind class and uses `build_modality_embeddings` and `build_output_modality_embeddings` to initialize embeddings. However I'm confused about what to do as the generation workflows is very different from the finetuning one. 

Could you help me figure out what I need to do ?

I think my question is different from https://github.com/IBM/terramind/issues/45

**Comment by blumenstiel** [comment]
Hi @fgiauna, why would u use the TM decoder if you finetune your model on burn scars? Short answer: It's not possible easily with TT. You can check the TerraMind Generation model and modify it to get the mae model with the right embeddings.

**Comment by fgiauna** [comment]
I'm working on a semi-supervised setting, therefore I need both encoder and decoder. 

If i use TerraMindGeneration can I - and should I - set output modalities like input modalities as I'm not looking to generate other modalities ? For instance on burn-scars would model_args be `"backbone_modalities": ["S2L2A"]` and `"backbone_output_modalities": ["S2L2A"],` ?

**Comment by blumenstiel** [comment]
you can do that, but the output will be tokens, not raw images if you only use the encoder decoder model without detokenizers.

**Comment by fgiauna** [comment]
I'm not sure I understood correctly what you meant with "detokenizers". Are you refering to what is done in the forward pass of TerraMindGeneration (in terramind_generation.py at lines 343-348) ? :
```
out[self.output_mod_name_mapping[mod]] = self.tokenizer[mod].decode_tokens(
                    tok,
                    image_size=image_size,
                    timesteps=timesteps,
                    verbose=verbose
                )
```

**Comment by blumenstiel** [comment]
Yes exactly. The Enc Dec model predicts tokens, not raw pixels. These tokens are then decoded by tokenizers. The 4M paper describes this process quite detailed. So you 1. need to decide which pretraining modality to predict for your task, and 2. you need some additional layers to produce the prediction mask. I am not sure if this would work better than standard decoder architectures.
